### PR TITLE
[Improve](serde)update serialize and deserialize text for data type

### PR DESCRIPTION
**Pull request by amorynan**
## Proposed changes
this pr is for centralize text serde for data types , now we use different from_string and to_string in field, wrapper_field, data_types/ and predict_column ... , but serde at text can be unified for specify data type, not scattered everywhere. 
Issue Number: close #xxx

In this pr :
1. from_string_test aim to comparing using "from_string" at data_type and wrapper field 
2. data_type_serde_text_test aim to comparing serde text action can equal to wrapper field in scala type 

after this pr, I will do
1. replace wrapper field with core/field and using data type serde instead of using  wrapper field from_string method
2. complete nested datatype (array/map/struct) text serde (support nested type in text) 
## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by amorynan** [comment]
run buildall

File: be/src/vec/data_types/serde/data_type_serde.h [hunk 1]
```diff
@@ -57,9 +59,20 @@ class IDataType;
 // the developer does not know how many datatypes has to deal.
 
 class DataTypeSerDe {
+public:
+    struct FormatOptions {
+        bool use_lib_format = false;
```
Comment (eldenmoon) [inline be/src/vec/data_types/serde/data_type_serde.h:64]: `use_lib_format` could you add some example to explain more details of this flag?

File: be/src/vec/data_types/serde/data_type_number_serde.cpp [hunk 2]
```diff
@@ -92,6 +95,63 @@ void DataTypeNumberSerDe<T>::write_column_to_arrow(const IColumn& column, const
...
+        T val = 0;
+        if (!read_float_text_fast_impl(val, rb)) {
+            return Status::InvalidArgument("parse number fail, string: '{}'",
+                                           std::string(rb.position(), rb.count()).c_str());
+        }
+        column_data.insert_value(val);
+    } else if constexpr (std::is_same_v<T, uint8_t>) {
+        // Note: here we should handle the bool type
+        T val = 0;
+        if (!try_read_bool_text(val, rb)) {
```
Comment (eldenmoon) [inline be/src/vec/data_types/serde/data_type_number_serde.cpp:115]: what if doris provide uint_8 type? The input  rb could be '123' and result Error?

File: be/src/vec/data_types/serde/data_type_number_serde.cpp [hunk 3]
```diff
@@ -92,6 +95,63 @@ void DataTypeNumberSerDe<T>::write_column_to_arrow(const IColumn& column, const
     }
 }
 
+template <typename T>
+Status DataTypeNumberSerDe<T>::deserialize_one_cell_from_text(IColumn& column, ReadBuffer& rb,
+                                                              const FormatOptions& options) const {
+    auto& column_data = reinterpret_cast<ColumnType&>(column);
+    if constexpr (std::is_same<T, UInt128>::value) {
```
Comment (eldenmoon) [inline be/src/vec/data_types/serde/data_type_number_serde.cpp:102]: why not handle float and double?

File: be/src/vec/data_types/serde/data_type_number_serde.cpp (hunk 2 above)
Comment (amorynan) [reply be/src/vec/data_types/serde/data_type_number_serde.cpp:115]: inner try_read_bool_text() we just make judge rb == 0 || rb == 1 , and return true or false , so will not result error, but here now we think uint_8 equals bool , just keep same with data_type_number_base.cpp from_string behavior, but I do not sure when I get a false ,like '123' , can we just insert into vector<uint8> column? is that much ambiguity for uint8 and bool ?

File: be/test/vec/data_types/from_string_test.cpp [hunk 4]
```diff
@@ -0,0 +1,162 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
```
Comment (yiguolei) [inline be/test/vec/data_types/from_string_test.cpp:9]: add some ut for column const , column nullable const

File: be/src/vec/data_types/serde/data_type_date64_serde.cpp [hunk 5]
```diff
@@ -23,10 +23,122 @@
...
+    row_num = result.second;
+
+    Int64 int_val = assert_cast<const ColumnInt64&>(*ptr).get_element(row_num);
+    if (options.use_lib_format) {
+        tm time_tm;
+        memset(&time_tm, 0, sizeof(time_tm));
+        time_tm.tm_mday = static_cast<int>(int_val & 31);
+        time_tm.tm_mon = static_cast<int>(int_val >> 5 & 15) - 1;
+        time_tm.tm_year = static_cast<int>(int_val >> 9) - 1900;
+        char buf[20] = {'\0'};
```
Comment (yiguolei) [inline be/src/vec/data_types/serde/data_type_date64_serde.cpp:45]: 这样写太废了，相当于这里先搞一个buf 数组，然后填充；再转string 这样会再copy 一次；然后buffer writable 会继续copy 一次

Comment (amorynan) [reply be/src/vec/data_types/serde/data_type_date64_serde.cpp:45]: just make same code with type.h to_string() , so not suggest use this if

**Comment by amorynan** [comment]
run buildall

**Review by eldenmoon — DISMISSED** [review]
LGTM

**Comment by amorynan** [comment]
run clickbench

**Comment by amorynan** [comment]
run buildall

**Comment by amorynan** [comment]
run buildall

**Comment by amorynan** [comment]
run p0

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 39.28 seconds
 stream load tsv:          468 seconds loaded 74807831229 Bytes, about 152 MB/s
 stream load json:         22 seconds loaded 2358488459 Bytes, about 102 MB/s
 stream load orc:          57 seconds loaded 1101869774 Bytes, about 18 MB/s
 stream load parquet:          29 seconds loaded 861443392 Bytes, about 28 MB/s
 insert into select:          71.1 seconds inserted 10000000 Rows, about 140K ops/s
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230704042042_clickbench_pr_171749.html

**Comment by amorynan** [comment]
run buildall

**Comment by amorynan** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 51.46 seconds
 stream load tsv:          500 seconds loaded 74807831229 Bytes, about 142 MB/s
 stream load json:         19 seconds loaded 2358488459 Bytes, about 118 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          30 seconds loaded 861443392 Bytes, about 27 MB/s
 insert into select:          28.6 seconds inserted 10000000 Rows, about 349K ops/s
 storage size: 17168518683 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230709214937_clickbench_pr_175234.html

**Comment by amorynan** [comment]
run clickbench

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 45.45 seconds
 stream load tsv:          450 seconds loaded 74807831229 Bytes, about 158 MB/s
 stream load json:         19 seconds loaded 2358488459 Bytes, about 118 MB/s
 stream load orc:          57 seconds loaded 1101869774 Bytes, about 18 MB/s
 stream load parquet:          30 seconds loaded 861443392 Bytes, about 27 MB/s
 insert into select:          25.7 seconds inserted 10000000 Rows, about 389K ops/s
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230709231602_clickbench_pr_175359.html

**Review by yiguolei — DISMISSED** [review]
LGTM

File: be/test/vec/data_types/from_string_test.cpp [hunk 6]
```diff
@@ -0,0 +1,177 @@
...
+                FieldType::OLAP_FIELD_TYPE_CHAR,       FieldType::OLAP_FIELD_TYPE_VARCHAR,
+                FieldType::OLAP_FIELD_TYPE_STRING,     FieldType::OLAP_FIELD_TYPE_DECIMAL,
+                FieldType::OLAP_FIELD_TYPE_DECIMAL32,  FieldType::OLAP_FIELD_TYPE_DECIMAL64,
+                FieldType::OLAP_FIELD_TYPE_DECIMAL128I};
+
+        for (auto type : arithmetic_scala_field_types) {
+            DataTypePtr data_type_ptr;
+            if (type == FieldType::OLAP_FIELD_TYPE_DECIMAL32) {
+                data_type_ptr = DataTypeFactory::instance().create_data_type(type, 9, 0);
+            } else if (type == FieldType::OLAP_FIELD_TYPE_DECIMAL64) {
```
Comment (yiguolei) [inline be/test/vec/data_types/from_string_test.cpp:52]: test is too simple。
For decimal type, please test decimalv2, decimalv3(decimal32,decimal64,decimal128) and the precision should be 0 or 2

Comment (yiguolei) [reply be/test/vec/data_types/from_string_test.cpp:52]: and datetimev2 also has pricision, should also test different precisions.

File: be/test/vec/data_types/from_string_test.cpp [hunk 7]
```diff
@@ -0,0 +1,177 @@
...
+#include "vec/io/reader_buffer.h"
+
+namespace doris::vectorized {
+
+/**
+ * This test is used to check wrapperField from_string is equal to data type from_string or not
+ *  same string feed to wrapperField and data type from_string, and check the result from
+ *  wrapperField and data type to_string is equal or not
+ */
+TEST(FromStringTest, ScalaWrapperFieldVsDataType) {
```
Comment (yiguolei) [inline be/test/vec/data_types/from_string_test.cpp:35]: std::string  decimal32_val = "32.12"
auto serde = createSerde(10,4);
Column col = serde.from_string(col, decimal32_val);
col.get_row_num(1) == 32.1200

std::string  decimal32_val = "32.1234567891011"
auto serde = createSerde(10,4);
Column col = serde.from_string(col, decimal32_val);
col.get_row_num(1) == 32.1200

serde = createserde(10,1)
col.get_row_num(1) == 32.1

std::string  decimal32_val = "32.16"
auto serde = createSerde(10,1);
col.get_row_num(1) == 32.2 ??


datetime
datetimev2 (precision) ?

Comment (yiguolei) [reply be/test/vec/data_types/from_string_test.cpp:35]: decimal32
decimal64
decimal128

decimalv2

Comment (yiguolei) [reply be/test/vec/data_types/from_string_test.cpp:35]: decimalv2 是固定的
decimalv2(27,9)

decimalv2(27,8) 的行为应该跟 decimalv2(27,9) 一样的

**Comment by amorynan** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 45.14 seconds
 stream load tsv:          506 seconds loaded 74807831229 Bytes, about 140 MB/s
 stream load json:         21 seconds loaded 2358488459 Bytes, about 107 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          29.3 seconds inserted 10000000 Rows, about 341K ops/s
 storage size: 17160224993 Bytes

**Comment by amorynan** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 46.18 seconds
 stream load tsv:          506 seconds loaded 74807831229 Bytes, about 140 MB/s
 stream load json:         21 seconds loaded 2358488459 Bytes, about 107 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          30 seconds loaded 861443392 Bytes, about 27 MB/s
 insert into select:          29.6 seconds inserted 10000000 Rows, about 337K ops/s
 storage size: 17169817249 Bytes

**Review by yiguolei — DISMISSED** [review]
LGTM

**Comment by amorynan** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 44.95 seconds
 stream load tsv:          504 seconds loaded 74807831229 Bytes, about 141 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          29.3 seconds inserted 10000000 Rows, about 341K ops/s
 storage size: 17161641918 Bytes

**Review by eldenmoon — APPROVED** [review]
LGTM

**Review by yiguolei — APPROVED** [review]
LGTM

**Pull request merged**